libclamav/CMakeLists.txt

```diff
@@ -568,6 +568,9 @@ if(ENABLE_SHARED_LIB)
     else()
         target_link_libraries( clamav PUBLIC Iconv::Iconv )
     endif()
+    if(UNIX)
```

Comment:
@opoplawski it appears this breaks the build for macOS with:
```
...
ld: unknown option: --version-script=/Users/runner/work/clamav/clamav/libclamunrar_iface/libclamunrar_iface.map
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [libclamunrar_iface/libclamunrar_iface.9.2.0.dylib] Error 1
make[1]: *** [libclamunrar_iface/CMakeFiles/clamunrar_iface.dir/all] Error 2
make: *** [all] Error 2
```

See: https://github.com/Cisco-Talos/clamav/actions/runs/3509775471/jobs/5879108732

Maybe we need to change to 

```
if(UNIX AND NOT APPLE)
```

Same for the other `if(UNIX)`'s in libclamunrar_iface/CMakeLists.txt and libfreshclam/CMakeLists.txt.

Reply:
Yeah, that's probably a good first-order approximation.  I suspect you'll see issues on other non-GNU platforms though.  I've pushed that change.